wallet/test/Test/Pos/Wallet/Web/AddressSpec.hs

```diff
@@ -30,34 +32,40 @@ import           Pos.Wallet.Web.State (askWalletSnapshot, getWalletAddresses,
                      wamAddress)
 import           Pos.Wallet.Web.Util (decodeCTypeOrFail)
 import           Test.Pos.Chain.Genesis.Dummy (dummyEpochSlots)
-import           Test.Pos.Configuration (withDefConfigurations)
+import           Test.Pos.Configuration (withDefConfigurations,
+                     withProvidedMagicConfig)
 import           Test.Pos.Util.QuickCheck.Property (assertProperty, expectedOne)
 import           Test.Pos.Wallet.Web.Mode (WalletProperty)
 import           Test.Pos.Wallet.Web.Util (importSingleWallet,
                      mostlyEmptyPassphrases)
 
 spec :: Spec
 spec = beforeAll_ setupTestLogging $
-            withDefConfigurations $ \_ _ _ ->
-                describe "Fake address has maximal possible size" $
-                modifyMaxSuccess (const 10) $ do
-                    prop "getNewAddress" $
-                        fakeAddressHasMaxSizeTest changeAddressGenerator
-                    prop "genUniqueAddress" $
-                        fakeAddressHasMaxSizeTest commonAddressGenerator
+    -- TODO @intricate: Not sure how to make the constraints play nice here
+    -- without `with*Configurations`.
+    withDefConfigurations $ \_ _ _ -> do
+        describe "Fake address has maximal possible size" $
+            modifyMaxSuccess (const 10) $ do
+                prop "getNewAddress" $ \pm ->
+                    withProvidedMagicConfig pm $ \genesisConfig _ _ ->
+                        fakeAddressHasMaxSizeTest changeAddressGenerator genesisConfig
+                prop "genUniqueAddress" $ \pm ->
+                    withProvidedMagicConfig pm $ \genesisConfig _ _ ->
+                        fakeAddressHasMaxSizeTest commonAddressGenerator genesisConfig
```

Comment:
`withProvidedMagicConfig` seems to shadow `withDefConfigurations` (for the constraints which `withProvidedMagicConfig` provides). We still need `withDefConfigurations`, though, because it provides some other constraints which we need. We're not sure why we need them - that discussion is out of scope at present.